mllib/src/main/scala/org/apache/spark/ml/feature/MinMaxScaler.scala

```diff
@@ -176,27 +177,47 @@ class MinMaxScalerModel private[ml] (
   @Since("2.0.0")
   override def transform(dataset: Dataset[_]): DataFrame = {
     transformSchema(dataset.schema, logging = true)
-    val originalRange = (originalMax.asBreeze - originalMin.asBreeze).toArray
+
+    val numFeatures = originalMax.size
+    val scale = $(max) - $(min)
+    val minValue = $(min)
+
+    // transformed value for constant cols
+    val constantOutput = 0.5 * scale + minValue
     val minArray = originalMin.toArray
 
-    val reScale = udf { (vector: Vector) =>
-      val scale = $(max) - $(min)
+    val scaleArray = new Array[Double](numFeatures)
+    var i = 0
+    while (i < numFeatures) {
```

Comment:
How much do we need this var / while construct in places like this, instead of functional programming? This isn't performance critical.

`val scaleArray = originalMax.zip(originalMin).map { case (max, min) => if (max == min) 0.0 else scale / (max - min) }`